state-chain/pallets/cf-ingress-egress/src/tests.rs

```diff
@@ -1193,8 +1188,23 @@ fn basic_balance_tracking() {
 		)])
 		.check_deposit_balances(&[
 			(eth::Asset::Eth, ETH_DEPOSIT_AMOUNT * 3),
+			// The ETH-FLIP swap uses the mock's hardcoded rate of 1:1.
 			(eth::Asset::Flip, FLIP_DEPOSIT_AMOUNT - ETH_DEPOSIT_AMOUNT),
 			(eth::Asset::Usdc, USDC_DEPOSIT_AMOUNT),
+		])
+		// Trigger a non-swap transfer.
+		.then_execute_at_next_block(|_| {
+			assert_ok!(<IngressEgress as EgressApi<Ethereum>>::schedule_egress(
+				eth::Asset::Flip,
+				FLIP_EGRESS_AMOUNT,
+				Default::default(),
+				None
+			));
+		})
+		.check_deposit_balances(&[
```

Comment:
I agree, could be more thorough, but this will require changing the ChainTracking mocks (since this is what 'delivers' the fee estimate). I'd rather leave this for another PR, this one has spun a little out of control already.

Reply:
(https://linear.app/chainflip/issue/PRO-1173/improve-chain-tracking-fee-estimation-mocks-and-tests)